#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Revert Jakub's change in 2003-04-02 to be enable to
# DP:		compile librt.so.
# DP: Author: GOTO Masanori <gotom@debian.org>
# DP: Upstream status: Not submitted
# DP: Status Details: under investigation.
# DP: Date: 2003-04-19

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- linuxthreads/Makefile	2003-04-16 10:00:17.000000000 +0900
+++ linuxthreads/Makefile.new	2003-04-18 09:37:24.000000000 +0900
@@ -47,12 +47,13 @@
 		       ptw-lseek64 ptw-llseek ptw-msync ptw-nanosleep \
 		       ptw-open ptw-open64 ptw-pause ptw-pread ptw-pread64 \
 		       ptw-pwrite ptw-pwrite64 ptw-tcdrain ptw-wait \
-		       ptw-waitpid pt-system old_pthread_atfork pthread_atfork
+		       ptw-waitpid pt-system old_pthread_atfork
+#		       ptw-waitpid pt-system old_pthread_atfork pthread_atfork
 # Don't generate deps for calls with no sources.  See sysdeps/unix/Makefile.
 omit-deps = $(unix-syscalls:%=ptw-%)
 
 libpthread-shared-only-routines = pt-allocrtsig
-libpthread-static-only-routines = pthread_atfork
+# libpthread-static-only-routines = pthread_atfork
 
 libpthread-nonshared = pthread_atfork
 
